src/object/obj_class.c

```diff
@@ -482,3 +496,76 @@ obj_ec_codec_get(daos_oclass_id_t oc_id)
 
 	return NULL;
 }
+
+
+/**
+ * Encode (using ISA-L) a full stripe from the submitted scatter-gather list.
+ *
+ * oid		[IN]		The object id of the object undergoing encode.
+ * sgl		[IN]		The SGL containing the user data.
+ * sg_idx	[IN|OUT]	Index of sg_iov entry in array.
+ * sg_off	[IN|OUT]	Offset into sg_iovs' io_buf.
+ * parity	[IN|OUT]	Struct containing parity buffers.
+ * p_idx	[IN]		Index into parity p_bufs array.
+ */
+int
+obj_encode_full_stripe(daos_obj_id_t oid, daos_sg_list_t *sgl, uint32_t *sg_idx,
+		       size_t *sg_off, struct obj_ec_parity *parity, int p_idx)
+{
+	struct obj_ec_codec		*codec = obj_ec_codec_get(
+							daos_obj_id2class(oid));
+	struct daos_oclass_attr		*oca = daos_oclass_attr_find(oid);
+	unsigned int			 len = oca->u.ec.e_len;
+	unsigned int			 k = oca->u.ec.e_k;
+	unsigned int			 p = oca->u.ec.e_p;
+	unsigned char			*data[k];
+	unsigned char			*ldata[k];
+	int				 i, lcnt = 0;
+	int				 rc = 0;
+
+	for (i = 0; i < k; i++)
+		if (sgl->sg_iovs[*sg_idx].iov_len - *sg_off >= len) {
```

Comment:
If the SGL's sg_iov array ends before the full stripe is accounted for, then the IOD specifies more data than is in the SGL (which is an error).  However, we don't verify the update request, so I probably should check for NULL.  I should probably verify the length of the iov_bufs as well, so I don't copy or encode garbage.  However, this would again be an error in the input (SGL should have as much data as IOD specifies).

Reply:
Ah this is only for pure total stripe?  hmm, let's say the cell_size is 1M, 4+2 object, and if it only write [0-3M], sgl include 3 buffers, each buffer has 1M. Then *sgl_idx in 535 line will keep increasing, when it gets to 3, then 527 line will be panic?

Reply:
this func is for full stripe

Reply:
minor personal preference to wrap equation in parens